wcfsetup/install/files/js/WoltLabSuite/Core/ImageResizer.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * This module allows resizing and conversion of HTMLImageElements to Blob and File objects
+ *
+ * @author	Maximilian Mader
+ * @copyright	2001-2018 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module	WoltLabSuite/Core/ImageResizer
+ */
+define([], function() {
+	/**
+	 * @constructor
+	 */
+	function ImageResizer() {
+		this.canvas = document.createElement('canvas');
+		this.ctx = this.canvas.getContext('2d');
+	}
+	
+	ImageResizer.prototype = {
+		maxWidth: 800,
+		maxHeight: 600,
+		quality: 0.8,
+		fileType: 'image/jpeg',
+
+		setMaxWidth: function (value) {
+			if (value == null) value = ImageResizer.prototype.maxWidth;
+			
+			this.maxWidth = value;
+			return this;
+		},
+		
+		setMaxHeight: function (value) {
+			if (value == null) value = ImageResizer.prototype.maxHeight;
+			
+			this.maxHeight = value;
+			return this;
+		},
+		
+		setQuality: function (value) {
+			if (value == null) value = ImageResizer.prototype.quality;
+			
+			this.quality = value;
+			return this;
+		},
+		
+		setFileType: function (value) {
+			if (value == null) value = ImageResizer.prototype.fileType;
+			
+			this.fileType = value;
+			return this;
+		},
+		
+		blobToFile: function (blob, name) {
+			var ext = '';
+			switch (blob.type) {
+				case 'image/png':
+					ext = '.png';
+					break;
+				
+				case 'image/jpeg':
+					ext = '.jpg';
+					break;
+				
+				case 'image/gif':
+					ext = '.gif';
+					break;
+				
+				default:
+					ext = blob.type.split('/')[1];
+			}
+			
+			return new File([blob], name.split(/(.+)(\..+?)$/)[1] + "_autoscaled" + ext);
```

Comment:
Using `match` probably is more intuitive:
```suggestion
			return new File([blob], name.match(/(.+)\..+?$/)[1] + "_autoscaled" + ext);
```